Revise first steps tutorial

tedinski opened the pull request:
### Description of changes: 

Revise "first steps." Notably:

1. Re-add the end of section summary/learning objectives.
2. Better define a few terms (proof harness, trace) before we use them.
3. Use `cargo kani` not `kani`
4. Small fixes

### Resolved issues:

toward #1320


### Call-outs:

<!-- 
Address any potentially confusing code. Is there code added that needs to be cleaned up later? Is there code that is missing because it’s still in development? 
-->

### Testing:

* How is this change tested?

* Is this a refactor change?

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

zhassan-aws commented on docs/src/tutorial-first-steps.md:
```diff
@@ -54,27 +55,37 @@ Notably, we haven't had to write explicit assertions in our proof harness: by de
 By default, Kani only reports failures, not how the failure happened.
-This is because, in its full generality, understanding how a failure happened requires exploring a full (potentially large) execution trace.
-Here, we've just got some nondeterministic inputs up front, but that's something of a special case that has a "simpler" explanation (just the choice of nondeterministic input).
+In this running example, it seems obvious what we're interested in (what `x`?) because we just have one unknown input at the start (similar to the property test), but that's something of a special case.
```
```suggestion
In this running example, it seems obvious what we're interested in (the value of `x` that caused the failure) because we just have one unknown input at the start (similar to the property test), but that's kind of a special case.
```

tedinski commented on docs/src/tutorial-first-steps.md:
```diff
@@ -2,7 +2,7 @@
 Kani is unlike the testing tools you may already be familiar with.
 Much of testing is concerned with thinking of new corner cases that need to be covered.
-With Kani, all the corner cases are covered from the start, and the new concern is narrowing down the scope to something manageable for the checker.
+With Kani, all the corner cases are covered from the start, and the new concern is narrowing down the scope to something manageable for the verifier.
```
I get what you're saying, but I liked that the original phrasing:

1. Avoids the word "correct", instead staying closer to the metaphor of testing.
2. Clearly indicates there's a tradeoff of sorts.

I've been trying to figure out a better phrasing, but it's taking me too long. Ok if I just leave this for now?

zhassan-aws replied:
Sure.

tedinski commented:
Updated.

Also, I noticed that we still had some functions named `main` around, and I fixed that too. This also required renaming `main.rs` to `lib.rs` in these examples.
